packages/flutter/test/material/snack_bar_test.dart

```diff
@@ -2291,4 +2291,36 @@ void main() {
 
     await expectLater(find.byType(MaterialApp), matchesGoldenFile('snack_bar.goldenTest.workWithBottomSheet.png'));
   });
+
+  testWidgets('ScaffoldMessenger presents SnackBars to only the root Scaffold when Scaffolds are nested.', (WidgetTester tester) async {
+    await tester.pumpWidget(MaterialApp(
+      home: Scaffold(
+        body: const Scaffold(),
+        floatingActionButton: FloatingActionButton(onPressed: () {}),
+      ),
+    ));
+
+    final ScaffoldMessengerState scaffoldMessengerState = tester.state(
```

Comment:
`tester.date<ScaffoldMessengerState>(`

It's a shame that the same identifier has to be repeated 3x :-(

Reply:
What is being repeated 3x?

Reply:
final **ScaffoldMessengerState** **scaffoldMessengerState** = tester.state<**ScaffoldMessengerState**>(